src/basics.md

```diff
@@ -270,6 +271,40 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-verify-extract-email]: #ex-verify-extract-email
+<a name="ex-verify-extract-email"></a>
+## Verify and extract login from an email address
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Validates that an email address is formatted correctly, and extracts everything
+before the @ symbol.
+
+```rust,no_run
+extern crate regex;
+use regex::Regex;
+
+// A simple email validator and extractor. Validates that something has an @
+// symbol, followed by something that looks like a domain.
```

Comment:
I would suggest to  move such descriptions to top with rest of the text. Also I would try to keep the comments as brief as possible (or even get rid of them in favour of textual description above) as we do not try to teach regular expressions here just show off the crate and snippet.